src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        if (index == 0 && head != null) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+        } else if (index == 0 && head == null) {
+            Node<T> newNode = new Node<>(null, value, null);
+            head = newNode;
+            tail = newNode;
+            head.next = tail;
+            tail.prev = head;
+        } else if (index == size) {
+            Node<T> newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        } else {
+            Node<T> currentNode = getNodeByIndex(index);
+            Node<T> previous = currentNode.prev;
+            Node<T> newNode = new Node<>(previous, value, currentNode);
+            previous.next = newNode;
+            currentNode.prev = newNode;
+        }
+        size++;
```

Comment:
lets simplify this method
```suggestion
    public void add(T value, int index) {
        checkAddIndex(index);
        if (index == size) {
             add(value);
             return;
        }
        Node<T> currentNode = getNodeByIndex(index);
            Node<T> previous = currentNode.prev;
            Node<T> newNode = new Node<>(previous, value, currentNode);
            previous.next = newNode;
            currentNode.prev = newNode;
        }
        size++;
```